crates/kornia-io/src/functional.rs

```diff
@@ -110,25 +144,32 @@ pub fn read_image_any(file_path: impl AsRef<Path>) -> Result<Image<u8, 3>, IoErr
     }
 
     // open the file and map it to memory
+    // TODO: explore whether we can use a more efficient memory mapping approach
     let file = std::fs::File::open(file_path)?;
     let mmap = unsafe { memmap2::Mmap::map(&file)? };
 
     // decode the data directly from memory
     // TODO: update the image crate
+    // TODO: explore supporting directly the decoders
     #[allow(deprecated)]
     let img = image::io::Reader::new(std::io::Cursor::new(&mmap))
         .with_guessed_format()?
         .decode()?;
 
-    // TODO: handle more image formats
-    // return the image data
-    let image = Image::new(
-        ImageSize {
-            width: img.width() as usize,
-            height: img.height() as usize,
-        },
-        img.to_rgb8().to_vec(),
-    )?;
+    let size = ImageSize {
+        width: img.width() as usize,
+        height: img.height() as usize,
+    };
+
+    let image = match img.color() {
+        image::ColorType::L8 => {
+            GenericImage::L8(Image::<u8, 1>::new(size, img.into_luma8().to_vec())?)
+        }
+        image::ColorType::Rgb8 => {
+            GenericImage::Rgb8(Image::<u8, 3>::new(size, img.into_rgb8().to_vec())?)
```

Comment:
One more change is `into_vec` (instead of `to_vec`) as noted in https://github.com/kornia/kornia-rs/issues/149#issuecomment-2380616277 . Just wanted to remind you in case you missed it. (I haven't tried it myself, so please ignore if it has any issues/negatives).

> * [`into_vec()`](https://docs.rs/image/latest/src/image/buffer.rs.html#1325-1330) consumes the image and gives us the inner vector without copy. `to_vec`, on the other hand, seems to point to [`slice::to_vec`](https://doc.rust-lang.org/1.81.0/src/alloc/slice.rs.html#449-451) (according to rust-analyzer) which copies.

Reply:
Yep, my mistake